src/utils/gitUtils.ts

```diff
@@ -63,10 +63,20 @@ export async function verifyGitWorkspaceForCreation(context: IActionContext, git
         gitWorkspaceState.repo = newRepo;
     } else if (!!gitWorkspaceState.remoteRepo && !gitWorkspaceState.hasAdminAccess) {
         context.telemetry.properties.cancelStep = 'adminAccess';
+        const adminAccess: string = localize('adminAccess', 'Admin access to the GitHub repository "{0}" is required. Would you like to create a fork?', gitWorkspaceState.remoteRepo.name);
+        const createForkItem: MessageItem = { title: localize('createFork', 'Create Fork') };
+        await ext.ui.showWarningMessage(adminAccess, { modal: true }, createForkItem);
 
-        const adminAccess: string = localize('adminAccess', 'Admin access to the GitHub repository is required.  Use a repo with admin access or create a fork.');
-        await ext.ui.showWarningMessage(adminAccess, { modal: true });
+        const repoUrl: string = (await createFork(context, gitWorkspaceState.remoteRepo)).data.html_url;
 
+        context.telemetry.properties.cancelStep = 'cloneFork';
+        const forkSuccess: string = localize('forkSuccess', 'Successfully forked "{0}". Would you like to clone your new repository?', gitWorkspaceState.remoteRepo.name);
+        const clone: MessageItem = { title: localize('clone', 'Clone') };
+        await ext.ui.showWarningMessage(forkSuccess, { modal: true }, clone);
+
+        context.telemetry.properties.cancelStep = 'afterCloneFork';
+        await commands.executeCommand('git.clone', repoUrl);
```

Comment:
Oh yeah, could use our `cloneRepo` command?  It effectively does just this, but if we add some additional behavior, would be nice to consolidate this logic.

I know I have a spot where I use `await commands.executeCommand('git.clone', repoUrl);` that I should probably refactor as well...

Reply:
This was the only other `git.clone` I found:

https://github.com/microsoft/vscode-azurestaticwebapps/blob/ee207dde2470020ae4de4973070a71ee46743f6e/src/utils/workspaceUtils.ts#L58

Does the git extension prompt for the repo to clone in that case?

Reply:
Yes, without a resourceUrl, it prompts up like so:

![image](https://user-images.githubusercontent.com/5290572/117494044-a4996100-af28-11eb-9e1e-057fae65da9b.png)